snuba/state/rate_limit.py

```diff
@@ -267,6 +267,28 @@ def get_global_rate_limit_params() -> RateLimitParameters:
     )
 
 
+def _record_metrics(
+    exc: RateLimitExceeded, rate_limit_param: RateLimitParameters
+) -> None:
+    """
+    Record rate limit metrics if needed.
+
+    We only record the metrics for global and table rate limits since
+    those indicate capacity of clickhouse clusters.
+
+    We get the scope and name from the message of RateLimitExceeded
+    exception since we want to know whether we exceeded the concurrent
+    or per second limit.
+    """
+    scope, name = exc.message.split()[:2]
```

Comment:
Relying on the exception message doesn't seem a like a good idea. I would pass that information directly into the function.

Reply:
I also didn't want to rely on it initially.

The other way I was doing it initially is by raising different exception types in the code here https://github.com/getsentry/snuba/blob/master/snuba/state/rate_limit.py#L199-L228. But the problem is that the code is agnostic to the type of rate-limiting it is currently running. So I wanted to keep it that way. That meant passing the exception to raise in the `rate_limit_params`. Then the exception raised would also need to be different based on whether it is per-second or concurrent. So it starts getting complicated.

So I went with this mechanism. I am open to other ideas of achieving this without doing a lot of refactoring.

Reply:
Would it be cleaner to make separate attributes for the scope and the name on `RateLimitExceeded`? Maybe they could be optional if not every caller uses them

Reply:
Made the changes as per the suggestion